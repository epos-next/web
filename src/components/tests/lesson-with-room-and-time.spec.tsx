import LessonWithRoomAndTime from "@components/lesson-with-room-and-time";
import React from "react";
import { render } from "@testing-library/react";

const props = {
    room: "202",
    time: "11:00 - 12:00",
    subject: "Физика"
}

it("should match snapshot", () => {
    const { asFragment } = render(<LessonWithRoomAndTime {...props}/>)
    expect(asFragment()).toMatchSnapshot()
});

it("should use room prop", () => {
    const { getByTestId, rerender } = render(<LessonWithRoomAndTime {...props} />)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("202")

    rerender(<LessonWithRoomAndTime {...props} room="101"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("101")

    rerender(<LessonWithRoomAndTime {...props} room="1234567890"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("1234567890")
})

it("should use time prop", () => {
    const { getByTestId, rerender } = render(<LessonWithRoomAndTime {...props} />)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("11:00 - 12:00")

    rerender(<LessonWithRoomAndTime {...props} time="9:00 - 15:00"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("9:00 - 15:00")

    rerender(<LessonWithRoomAndTime {...props} time="valabuday"/>)
    expect(getByTestId("lesson-subtitle")).toHaveTextContent("valabuday")
})

it("should have \"lesson-with-room-and-time\" class", () => {
    const { getByTestId } = render(<LessonWithRoomAndTime {...props} />)
    expect(getByTestId("lesson-container")).toHaveClass("lesson-with-room-and-time")
});

it("should override default \"lesson-with-room-and-time\" class", () => {
    const { getByTestId } = render(<LessonWithRoomAndTime {...props} className="hello-123" />)
    expect(getByTestId("lesson-container")).toHaveClass("hello-123")
})

it("should show subject", () => {
    const subject = "Физика"
    const { getByTestId } = render(<LessonWithRoomAndTime {...props} subject={subject}/>)
    expect(getByTestId("lesson-name")).toHaveTextContent(subject)
});

