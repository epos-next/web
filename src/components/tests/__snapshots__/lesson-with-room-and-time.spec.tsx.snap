// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  .c4 {
  font-size: 17px;
  line-height: 20px;
  color: var(--primary);
}

.c5 {
  font-size: 15px;
  line-height: 18px;
  color: var(--secondary);
}

.c2 {
  width: 36px;
  height: 36px;
  border-radius: 50%;
  background-color: #BBEDBF;
  color: #68D676;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: 500;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 15px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  overflow: hidden;
}

.c7 {
  width: 3px;
  height: 3px;
  background: #696969;
  border-radius: 50%;
  margin: 0 10px;
}

<div
    class="c0"
  >
    <div
      class="c1 lesson-with-room-and-time"
      data-testid="lesson-container"
    >
      <div
        class="c2"
        color="#BBEDBF"
        data-testid="avatar"
      >
        Ф
      </div>
      <div
        class="c3"
      >
        <span
          class="c4"
          data-testid="lesson-name"
        >
          Физика
        </span>
        <span
          class="c5"
          data-testid="lesson-subtitle"
        >
          <div
            class="c6"
          >
            202
            <div
              class="c7"
            />
            11:00 - 12:00
          </div>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
